lib/ansible/plugins/lookup/passwordstore.py

```diff
@@ -35,20 +38,21 @@ def check_output2(*popenargs, **kwargs):
     if 'input' in kwargs:
         if 'stdin' in kwargs:
             raise ValueError('stdin and input arguments may not both be used.')
-        inputdata = kwargs['input']
+        inputdata = to_bytes(kwargs['input'])
         del kwargs['input']
         kwargs['stdin'] = subprocess.PIPE
     else:
         inputdata = None
     process = subprocess.Popen(*popenargs, stdout=subprocess.PIPE, stderr=subprocess.PIPE, **kwargs)
     try:
         out, err = process.communicate(inputdata)
+        out, err = to_native(out), to_native(err)
     except:
         process.kill()
         process.wait()
         raise
     retcode = process.poll()
-    if retcode:
+    if retcode != 0 or 'encryption failed: Unusable public key' in out + err:
         cmd = kwargs.get("args")
         if cmd is None:
             cmd = popenargs[0]
```

Comment:
If you follow my suggestion to leave out and err as byte strings, the two lines below this comment should be:
``` python
         raise subprocess.CalledProcessError(retcode, cmd, to_native(b_out + b_err))
return b_out
```